lib/benchee/statistics/percentile.ex

```diff
@@ -1,50 +1,67 @@
+  @type percentiles :: %{optional(number()) => percentile()}
+
   @doc """
-  Calculates the value at the `percentile_number`-th percentile. Think of this as the
-  value below which `percentile_number` percent of the samples lie. For example,
+  Calculates the value at the `percentile_rank`-th percentile. Think of this as the
+  value below which `percentile_rank` percent of the samples lie. For example,
   if `Benchee.Statistics.Percentile.percentile(samples, 99)` == 123.45,
   99% of samples are less than 123.45.
 
-  ## Examples
-
-  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, 100)
-  5.0
-
-  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, 150)
-  5.0
+  Passing a number for `percentile_rank` calculates a single percentile.
+  Passing a list of numbers calculates multiple percentiles, and returns them
+  as a map like %{90 => 45.6, 99 => 78.9}, where the keys are the percentile
+  numbers, and the values are the percentile values.
 
-  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, 0)
-  1.0
+  ## Examples
 
-  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, -1)
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 12.5)
   1.0
 
   iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 50)
   3.0
 
   iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 75)
   4.75
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 99)
+  5.0
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], [50, 75, 99])
+  %{50 => 3.0, 75 => 4.75, 99 => 5.0}
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 100)
+  ** (ArgumentError) percentile must be between 0 and 100, got: 100
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 0)
+  ** (ArgumentError) percentile must be between 0 and 100, got: 0
   """
-  @spec percentile(list(number()), integer()) :: float()
-  def percentile(samples, percentile_number) do
-    percentile(samples, length(samples), percentile_number)
+  @spec percentile(list(number()), number() | list(number())) ::
+    percentile | percentiles
+  def percentile(samples, percentile_ranks) do
+    number_of_samples = length(samples)
+    sorted = Enum.sort(samples)
+    percentile(sorted, number_of_samples, percentile_ranks)
   end
 
-  @spec percentile(list(number()), integer(), integer()) :: float()
-  def percentile(samples, number_of_samples, percentile_number) when percentile_number > 100 do
-    percentile(samples, number_of_samples, 100)
+  defp percentile(sorted_samples, number_of_samples, percentile_ranks)
+    when is_list(percentile_ranks) do
+    percentile_ranks
+    |> Enum.reduce(%{}, fn percentile_rank, acc ->
+      perc = percentile(sorted_samples, number_of_samples, percentile_rank)
+      Map.put(acc, percentile_rank, perc)
+    end)
```

Comment:
This is just one pipe, so I think it can be written a bit more easily as `Enum.reduce(percentile_ranks, %{} ...`

Reply:
CREDO!!! Fails on the line length check. Can't get that one thru CI

Reply:
Huh, I see you like credo :D I might come around on not enforcing 80 char width but not quite yet :D 

Worst case I'd break the arguments to multiple lines.

Reply:
It's going to be 98 characters with the formatter. That's one of the few things I disagree with, but c'est la vie! I'd be ok with changing Credo's line length check to be 98 characters now, since we know that's in the future anyway.

Reply:
98? 98????? Not 100. No 98? _exahles slowly_

(for people who don't know me, I'm mostly just joking around here but am genuinely confused)

Go ahead @devonestes

Reply:
I actually really like credo. But the hard 80-char limit is really tough in certain cases, like when you are trying to put a url in a comment

Reply:
_/me always uses a 120 char limit everywhere for everything, so this 98 limit in the new formatter is blegh_  ^.^;